components/camera/camera.go

```diff
@@ -119,6 +119,9 @@ type VideoSource interface {
 	// that may have a MIME type hint dictated in the context via gostream.WithMIMETypeHint.
 	Stream(ctx context.Context, errHandlers ...gostream.ErrorHandler) (gostream.VideoStream, error)
 
+	// GetImage returns a single image that may have a MIME type hint dictated in the context via gostream.WithMIMETypeHint.
+	GetImage(ctx context.Context) (image.Image, func(), error)
```

Comment:
Hm, dont we want to keep Get prefix to match the proto api?

Reply:
No, look at all our other [APIs](https://github.com/viamrobotics/api/blob/23fec5b989bfe3d95fbc4271ca42b10a7e71ed0c/proto/viam/component/camera/v1/camera.proto#L46) and how they're [wrapped in go](https://github.com/viamrobotics/rdk/blob/9c30b69187382b3dc1bda58069ad01a18f46acef/components/camera/server.go#L262). We drop the `Get` prepend because there is no getter/setter [idiom in go](https://go.dev/doc/effective_go#Getters), unlike our other sdks. So we drop the prepend in rdk because that's the style that people that are opinionated about this have followed. 

We could have not done this, but we should keep the style now that we're in deep.

Reply:
Oh lord there is an extra field

Reply:
But the request msg struct also contains camera name and mime type

https://github.com/viamrobotics/api/blob/23fec5b989bfe3d95fbc4271ca42b10a7e71ed0c/proto/viam/component/camera/v1/camera.proto#L67

Reply:
Don't worry about the extra field, it'll actually be better to make it explicit, because right now we're [ funnelling it through the ctx ](URL) in the server

Reply:
The name is unnecessary to include in the inputs too, look at the APIs client/server for any other component (motor is clean), the rpc resource name is not included in the function signature, it's provided by `Resource.Name`

Reply:
Wait so should we change it to `Image(ctx context.Context, mimeType string, extra interface{}) ([]byte, string, error)`? Or is it okay to leave everything it in the context?

Reply:
^ We should change it to Image(ctx, mimeType, extra); the reason extra is passed through the context is that gostream's `Stream` function signature does not have an extra and can't propagate it to the lower gostream functions.

... which we now no longer care about.

Reply:
It takes a lot of effort to go through and change the `Image` signature, so I want to verify that `Image(ctx context.Context, mimeType string, extra *structpb.Struct) ([]byte, string, error)` looks good before I do it

Reply:
In the client/server code we deserialize the extra, https://github.com/hexbabe/sean-rdk/blob/b8d7747a731116787cde0ae062ac6ee3571f5075/components/movementsensor/client.go#L109

But in the associated interface function it should be a map[string]interface https://github.com/hexbabe/sean-rdk/blob/b8d7747a731116787cde0ae062ac6ee3571f5075/components/movementsensor/movementsensor.go#L140

Reply:
Dug around and found the camera package's `Extra` type for extras. So prob gonna use that `Image(ctx context.Context, mimeType string, extra Extra) ([]byte, string, error)`

Reply:
> It takes a lot of effort to go through and change the Image signature, so I want to verify that Image(ctx context.Context, mimeType string, extra *structpb.Struct) ([]byte, string, error) looks good before I do it

I like this signature since it's 1:1 to our APIs and follows the other sdk's signature, would you like it to go through a scope though?

Reply:
A scope is only required for breaking proto changes right? This is a go interface not proto, so maybe not? Unless we feel like as a team it's worth getting more eyes on it.

Also what you quoted is the `extra *structpb.Struct` and not the `extra Extra` using the `Extra` type alias for `map[string]interface{}` in the `camera` package. This is intentional right? just wanted to clarify

Reply:
^ I do not think anything about stream and next are intentional, is that type alias used anywhere else in the camera package?